platform/security/sts/security-sts-ldaplogin/src/test/java/ddf/ldap/ldaplogin/LdapModuleTest.java

```diff
@@ -0,0 +1,214 @@
+/**
+ * Copyright (c) Codice Foundation
+ * <p>
+ * This is free software: you can redistribute it and/or modify it under the terms of the GNU Lesser
+ * General Public License as published by the Free Software Foundation, either version 3 of the
+ * License, or any later version.
+ * <p>
+ * This program is distributed in the hope that it will be useful, but WITHOUT ANY WARRANTY; without
+ * even the implied warranty of MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU
+ * Lesser General Public License for more details. A copy of the GNU Lesser General Public License
+ * is distributed along with this program and can be found at
+ * <http://www.gnu.org/licenses/lgpl.html>.
+ */
+
+package ddf.ldap.ldaplogin;
+
+import static org.hamcrest.MatcherAssert.assertThat;
+import static org.hamcrest.Matchers.containsInAnyOrder;
+import static org.hamcrest.Matchers.empty;
+import static org.junit.Assert.fail;
+import static ddf.ldap.ldaplogin.SslLdapLoginModule.BIND_METHOD;
+import static ddf.ldap.ldaplogin.SslLdapLoginModule.CONNECTION_PASSWORD;
+import static ddf.ldap.ldaplogin.SslLdapLoginModule.CONNECTION_URL;
+import static ddf.ldap.ldaplogin.SslLdapLoginModule.CONNECTION_USERNAME;
+import static ddf.ldap.ldaplogin.SslLdapLoginModule.KDC_ADDRESS;
+import static ddf.ldap.ldaplogin.SslLdapLoginModule.REALM;
+import static ddf.ldap.ldaplogin.SslLdapLoginModule.ROLE_BASE_DN;
+import static ddf.ldap.ldaplogin.SslLdapLoginModule.ROLE_FILTER;
+import static ddf.ldap.ldaplogin.SslLdapLoginModule.ROLE_NAME_ATTRIBUTE;
+import static ddf.ldap.ldaplogin.SslLdapLoginModule.ROLE_SEARCH_SUBTREE;
+import static ddf.ldap.ldaplogin.SslLdapLoginModule.SSL_STARTTLS;
+import static ddf.ldap.ldaplogin.SslLdapLoginModule.USER_BASE_DN;
+import static ddf.ldap.ldaplogin.SslLdapLoginModule.USER_FILTER;
+import static ddf.ldap.ldaplogin.SslLdapLoginModule.USER_SEARCH_SUBTREE;
+
+import java.security.Principal;
+import java.util.HashMap;
+import java.util.Map;
+import java.util.Set;
+
+import javax.security.auth.Subject;
+import javax.security.auth.callback.CallbackHandler;
+import javax.security.auth.callback.NameCallback;
+import javax.security.auth.callback.PasswordCallback;
+import javax.security.auth.login.LoginException;
+
+import org.apache.karaf.jaas.boot.principal.RolePrincipal;
+import org.apache.karaf.jaas.boot.principal.UserPrincipal;
+import org.junit.After;
+import org.junit.Before;
+import org.junit.Rule;
+import org.junit.Test;
+import org.junit.contrib.java.lang.system.ProvideSystemProperty;
+
+import com.github.trevershick.test.ldap.LdapServerResource;
+import com.github.trevershick.test.ldap.annotations.LdapAttribute;
+import com.github.trevershick.test.ldap.annotations.LdapConfiguration;
+import com.github.trevershick.test.ldap.annotations.LdapEntry;
+
+//Create and populate LDAP server
+// @formatter:off
+@LdapConfiguration(
```

Comment:
I didn't get tagged because it's in the code block for some reason. 😄 

I do not like disabling the formatter. I would prefer to reenable it, and if you feel the annotations look too ugly after they have been formatted, that's your answer.